posts/2024-05-01-spring-boot-3.adoc

```diff
@@ -0,0 +1,232 @@
+
+    @GetMapping("/")
+    public String hello() {
+        return "Hello from Spring Boot in Open Liberty!";
+    }
+
+}
+----
++
+3. Run with standard WAR deployment with default configuration
++
+[source,sh]
+----
+./mvnw io.openliberty.tools:liberty-maven-plugin:3.10:dev -Dtemplate=webProfile10 -DskipTests
+----
++
+4. Test the application by visiting the following URL in a browser: http://localhost:9080/spring-liberty-0.0.1-SNAPSHOT/
++
+You should see:
++
+  Hello from Spring Boot in Open Liberty!
++
+5. Stop the server (with  `<Ctrl>+C` or `q,<Enter>` to quit dev mode).
++
+6. Recap
++
+The app is bootstrapped by the web container via the ServletInitializer, extending `SpringBootServletInitializer`.  We go out of our way to run this example with only default Liberty and Maven configuration even though it'd be more typical to maintain a server.xml configuration for your app.
+
+== Use optimized Liberty Spring Boot deployment
+
+We're going to reuse the app from the previous example.   By adding some extra configuration in the `pom.xml` and `server.xml` we end up being able to build an efficiently-layered Docker image, and also to parameterize the app with application arguments.
+
+1. Add liberty-maven-plugin with Spring Boot deployment config to pom.xml:
++
+[source,xml]
+----
+    <plugin>
+        <groupId>io.openliberty.tools</groupId>
+        <artifactId>liberty-maven-plugin</artifactId>
+        <version>3.10</version>
+        <configuration>
+            <appsDirectory>apps</appsDirectory>
+            <deployPackages>spring-boot-project</deployPackages>
+        </configuration>
+    </plugin>
+----
++
+2. Create your Liberty server.xml by copying from the Spring Boot template
++
+[source,sh]
+----
+./mvnw clean liberty:create -Dtemplate=springBoot3
+mkdir -p src/main/liberty/config/
+cp target/liberty/wlp/usr/servers/defaultServer/server.xml src/main/liberty/config/server.xml
+----
++
+3. Add application arguments to your Spring Boot application configuration in Liberty server.xml
++
+[source,xml]
+----
+ <springBootApplication id="spring-liberty" location="thin-spring-liberty-0.0.1-SNAPSHOT.war" name="spring-liberty">
+    <applicationArgument>--server.servlet.context-path=/testpath</applicationArgument>
+    <applicationArgument>--myarg=myval</applicationArgument>
+ </springBootApplication>
+----
++
+4. Add println to echo args in output, `vi ./src/main/java/demo/SpringLibertyApplication.java`
++
+[source,java]
+----
+    public static void main(String[] args) {
+        System.out.println("Arguments array = " + java.util.Arrays.toString(args));
+        SpringApplication.run(SpringLibertyApplication.class, args);
+    }
+----
++
+5. Run
++
+Though the Liberty Maven Plugin doesn't currently have support for using dev mode (or "loose applications") when using the optimized deployment, we can still use the "lifecycle" built into the 'run' goal.  We do need to do a 'package' first to get the Spring Boot "repackage" packaging.
```

Comment:
```suggestion
Although the Liberty Maven Plugin doesn't currently support dev mode when you use the optimized deployment, you can still use the life cycle built into the 'run' goal.  We do need to do a 'package' first to get the Spring Boot "repackage" packaging.
```

Reply:
Took this but kept "lifecycle" as one word since it's meant to track the Maven term "lifecycle".